source/common/stats/thread_local_store.cc

```diff
@@ -940,6 +940,29 @@ const std::string ParentHistogramImpl::bucketSummary() const {
   }
 }
 
+const std::string ParentHistogramImpl::nonOverlappingBucketSummary() const {
+  if (used()) {
+    std::vector<std::string> bucket_summary;
+    ConstSupportedBuckets& supported_buckets = interval_statistics_.supportedBuckets();
+    bucket_summary.reserve(supported_buckets.size());
+    uint64_t previous_computed_interval_bucket = 0;
+    uint64_t previous_computed_cumulative_bucket = 0;
+    for (size_t i = 0; i < supported_buckets.size(); ++i) {
+      uint64_t current_computed_interval_bucket = interval_statistics_.computedBuckets()[i];
+      uint64_t current_computed_cumulative_bucket = cumulative_statistics_.computedBuckets()[i];
+      bucket_summary.push_back(
+          fmt::format("B{:g}({},{})", supported_buckets[i],
+                      current_computed_interval_bucket - previous_computed_interval_bucket,
+                      current_computed_cumulative_bucket - previous_computed_cumulative_bucket));
+      previous_computed_interval_bucket = current_computed_interval_bucket;
+      previous_computed_cumulative_bucket = current_computed_cumulative_bucket;
+    }
+    return absl::StrJoin(bucket_summary, " ");
```

Comment:
I feel like we should have this API return the array, and have a separate layer that joins it as strings.

For example, if I have a richer HTML display of histograms I might like to be able to render these values graphically.

Reply:
Moreover, I think it'd be great to return a structured result that's an array of triples, and do the formatting in the stats handler (e.g. json-population vs emitting strings manually)

Reply:
I'll start looking into this

Reply:
I added nonoverlappingComputedBuckets() to HistogramStatistics which can be used from the ParentHistogram. It returns an array with the desired values as output. Does this accomplish what you meant? I changed the nonoverlappingBucketSummary() to use the new method. Should I also move the summary to the stats handler? Originally I wrote it here because it is similar to the existing bucketSummary().

Reply:
This is better but I still think the formatting of this could be removed from the histogram and go into the stats handler class.

E.g. one thing I was thinking (in a follow-up after an in-progress PR) is to have HTML-mode actually render the histograms graphically. So the more we do via structured API, and the less we commit to string representations inside the Histogram class, the more flexible we are about how to present the data in the stats handler.

Reply:
I moved `computeDisjointBucketSummary()` to the stats handler.